File: 3656-minimum-number-of-operations-to-make-elements-in-array-distinct/3656-minimum-number-of-operations-to-make-elements-in-array-distinct.java
class Solution {
    public int minimumOperations(int[] nums) {
        int n = nums.length;
        boolean[] vis = new boolean[101];

        for (int i = n - 1; i >= 0; i--) {
            if (vis[nums[i]])
                return (int)Math.ceil((double)(i + 1) / 3);
            vis[nums[i]] = true;
        }
        return 0;
    }
}

File: 3656-minimum-number-of-operations-to-make-elements-in-array-distinct/3656-minimum-number-of-operations-to-make-elements-in-array-distinct.cs
public class Solution {
    public int MinimumOperations(int[] nums) {
        int n = nums.Length;
        var map = new bool[101];
        for (int i = n - 1; i >= 0; --i) {
            if (map[nums[i]])
                return (int)Math.Ceiling((double) (i + 1) / 3);
            map[nums[i]] = true;
        }
        return 0;
    }
}